src/adapter/ezsp/driver/ezsp.ts

```diff
@@ -271,10 +272,11 @@ export class Ezsp extends EventEmitter {
                 debug.error(`Connection attempt ${i} error: ${error.stack}`);
                 await Wait(5000);
                 debug.log(`Next attempt ${i+1}`);
+                lastError = error;
             }
         }
         if (!this.serialDriver.isInitialized()) {
-            throw new Error("Failure to connect");
+            throw new Error("Failure to connect", {cause: lastError});

```

Comment:
Not strictly necessary since it's in the logs already, but let me know if you'd like me to remove.

Reply:
It's fine